fili-core/src/test/groovy/com/yahoo/bard/webservice/data/metric/TemplateDruidQueryMergerSpec.groovy

```diff
@@ -65,17 +84,39 @@ class TemplateDruidQueryMergerSpec extends Specification {
             merged.depth() == 2
     }
 
+    def "if require metrics in queries is turned ON, merger.merge throws exception for empty list of metric"() {
+        setup:
+        SystemConfigProvider.getInstance().setProperty(AbstractBinderFactory.REQUIRE_METRICS_IN_QUERY_KEY, "true")
+
+        DataApiRequest request = Mock(DataApiRequest)
+        TemplateDruidQueryMerger merger = new TemplateDruidQueryMerger()
 
-    def "Verify merger.merge throws exception for empty list of metric"() {
+        when:
+        merger.merge(request)
+
+        then:
+        (1 .. _) * request.getLogicalMetrics() >> [].toSet()
+        thrown(IllegalStateException)
+
+    }
+
+    def "If require metrics in queries is turned OFF, a merge with no metrics will return an empty TDQ"() {
         setup:
-            DataApiRequest request = Mock(DataApiRequest)
-            TemplateDruidQueryMerger merger = new TemplateDruidQueryMerger()
+        SystemConfigProvider.getInstance().setProperty(AbstractBinderFactory.REQUIRE_METRICS_IN_QUERY_KEY, "false")
```

Comment:
I recommend you use FeatureFlag to manage this property in tests.
check out BardFeatureFlag